### PR TITLE
Update to rails 7.2

### PR DESCRIPTION
**Pull request by christinach**
Update marc_cleanup to v0.9.3
See release notes: https://github.com/pulibrary/marc_cleanup/releases/tag/v0.9.3
require marc_cleanup in scsb partner_updates
    
Don't require marc_cleanup: set require: false in the Gemfile for marc_cleanup because of the following error:
```
    NameError: undefined method `map' for module `#<Module:0x000000012dd5b0f0>' (NameError)
    
              owner.define_method(as, @cache.instance_method(canonical_name))
```

### CODE REVIEW

File: spec/requests/barcode_gets_spec.rb [hunk 1]
```diff
@@ -3,18 +3,16 @@
...
     end
   end
 
   describe 'GET /barcode/321010702214' do
-    it 'returns an error when the barcode is not a valid form' do
-      pending 'Replace with Alma'
-      # allow(VoyagerHelpers::Liberator).to receive(:get_records_from_barcode).and_return([])
+    it 'returns a 404 when the barcode is not a valid form', pending: 'Replace with Alma' do
+      # skipping because while pending it fails and breaks CI
+      skip
```
Comment (christinach) [inline spec/requests/barcode_gets_spec.rb:15]: Skipping this 'pending' test because it's breaking CI.

File: spec/requests/barcode_gets_spec.rb [hunk 2]
```diff
@@ -27,8 +25,3 @@
     end
   end
 end
-
-def stub_voyager_barcodes(barcode)
```
Comment (christinach) [inline spec/requests/barcode_gets_spec.rb:31]: We don't use voyager

File: spec/requests/barcode_gets_spec.rb [hunk 3]
```diff
@@ -3,18 +3,16 @@
 
 RSpec.describe 'Barcode Gets', type: :request do
   describe 'GET /barcode/32101070300312' do
-    it 'returns a collection of bibliographic records' do
-      pending 'Replace with Alma'
-      # stub_voyager_barcodes('32101070300312')
+    it 'returns a collection of bibliographic records', pending: 'Replace with Alma' do
```
Comment (christinach) [inline spec/requests/barcode_gets_spec.rb:6]: The pending specs in this file need to be reviewed and updated. I will create a ticket

**Review by sandbergja — APPROVED** [review]
Thanks, @christinach !  🐬

**Pull request merged**